reactor-core/src/jcstress/java/reactor/core/publisher/FluxPublishOnStressTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.core.publisher;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.Arbiter;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.I_Result;
+import org.reactivestreams.Subscription;
+
+import reactor.core.scheduler.Schedulers;
+import reactor.util.concurrent.Queues;
+import reactor.util.context.Context;
+
+import static org.openjdk.jcstress.annotations.Expect.*;
+
+abstract class FluxPublishOnStressTest {
+
+	@JCStressTest
+	//note: -1 can still happen but should be present in a very low number
+	@Outcome(id = {"-1"}, expect = ACCEPTABLE_INTERESTING, desc = "Latch timed out")
+	@Outcome(id = {"1"}, expect = ACCEPTABLE,  desc = "Simple discard")
+	@Outcome(id = {"10"}, expect = ACCEPTABLE,  desc = "Processed")
+	@Outcome(id = {"1001"}, expect = ACCEPTABLE_INTERESTING, desc = "Discarded and onError")
```

Comment:
What kind of error may happen there leading to the element being discarded. Is it something related to the scheduler error? Should not we just add onErrorDropped hook to ensure this scenario is related to rejected scheduler task